lib/features/destination_picker/presentation/destination_picker_controller.dart

```diff
@@ -0,0 +1,94 @@
+
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/base/base_controller.dart';
+import 'package:tmail_ui_user/features/destination_picker/presentation/model/destination_picker_arguments.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/state/get_all_mailboxes_state.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/usecases/get_all_mailbox_interactor.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_node.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_tree.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_tree_builder.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/extensions/list_mailbox_node_extension.dart';
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/main/routes/route_navigation.dart';
+
+class DestinationPickerController extends BaseController {
```

Comment:
please check the comment: https://github.com/linagora/tmail-flutter/pull/115#issuecomment-924124966. For this picker, we don't need to geed it from the server.

Reply:
Please set up the destination picker with the model: `in/out`
```
mailboxes -> user interaction in destination picker -> done button -> the picked mailbox send back to the invoker -> mailbox dashboard will execute the moving task -> toast
```
with this, we can reuse the destination picker in many places

Reply:
I don't understand what you say. Call me to speak more clearly.